pycqed/analysis_v2/base_analysis.py

```diff
@@ -252,7 +252,7 @@ def get_data_from_timestamp_list(self):
             folder = a_tools.get_folder(timestamp)
             h5mode = self.options_dict.get('h5mode', 'r+')
             h5filepath = a_tools.measurement_filename(folder)
-            data_file = h5py.File(h5filepath, h5mode)
+            self.data_file = h5py.File(h5filepath, h5mode)
```

Comment:
I removed making data_file an attribute. This is useless because we close it at the end of this function, so you even if you can call class_instance.data_file, this will just give you a closed file. Why did you make it an attribute?

Reply:
It is closed only if an exception occurs, which I think is good. Otherwise, I think we should close it only at the end of the analysis (see e.g. line 5682 in timedomain analysis, but in fact I agree there it could also be in a try-except clause). 
To me being able to access the data_file during the analysis is useful to have access to anything in the hdf5 file and I think it should be possible in an easy way without having to reconstruct it with h5py. I for instance use it at line 5784 in time domain analysis.

If you think it is better not to make it an attribute, then I would suggest to create a wrapper of ba.get_hdf_param_value() that is not a static function, and that would create the required data_file object from self and then pass the object to get_hdf_param_value(). Do you think this would be better? In that way, whenever you want (also in the notebook), you could just do e.g. analysis_obj.get_hdf_value("Instrument settings/qb1", "T1") to get the value without needing to import h5py and create an data_file object. The disadvantage of this feature is that if you use it in a loop then it might slow you down a bit as you open and close the file each time, compared to having it as an attribute.

Reply:
Ok I see why you need it. However, you should use the params_dict and numeric_params for your use case in line 5784. That's how analysis_v2 was intended to be used, everything passed to params_dict and numeric_params are extracted and saved in raw_data_dict. See the init or RabiAnalysis for example of how to use this. I think this is nice because then we don't need access to the data_file throughout the methods. I think you should do it like this as well and remove the attribute self.data_file.

Reply:
Yes, I thought about using that method indeed. I decided not to because it relies on passing the qubit names, and I think that those should not be mandatory if they can be inferred from the channel map , or calibration points, etc. i.e. the default should be to analyze all qubits in the file. However, you can get those qb names only after extracting the data, so not in the init but after extract_data(). On the other hand, the numeric params have to be done in the init :D, so one cannot have both. 
But okay, it is not a critical thing to mandatorily pass the qubit_names, so I will use that if it makes more sense. I might still also create the other function I spoke about, as I think it could be useful even in the notebook to access the hdf5 file

Reply:
I'm not sure I understand. Looking at your class, you pass the qb_names in the init so you have access to them there, and that's where you create the params_dict/numeric_params (as is done in RabiAnalysis). Since prepare_plots is one of the last things you call, these params will already have been extracted and are in raw_data_dict. In RabiAnalysis I do exactly this: create params_dict in init, then just retrieve them in prepare_plots. And it looks to me like your use case is exactly the same.

Reply:
I realized now that your were talking in general, that it would be nice to not have to pass qb_names.

Reply:
Yes, I think I have now implemented something that respects both our ideas. It uses the numeric param by default but if qb_names is not specified then it uses the wrapper I have created. So in any case, the self.data_file isn't there anymore

Reply:
what do you think? (see commit [521dd3e ](https://github.com/QudevETH/PycQED_py3/pull/79/commits/521dd3e1e1245a3733c097e21d55e26596687dd5)